From 1d03a110ed1c6941b1e1f3dc11602d258edef1a5 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Tue, 7 May 2019 16:15:54 +0200
Subject: [PATCH 1168/1200] arm64: dts: cn9130: fix the pcie memory size

The PCIe MEM space size for CP0 PCIe0 is bigger than for other PCIe
interfaces - reflect it in device-tree description.

Change-Id: I3a7169816e943a560ef661e82aec46c12ad51d92
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/10505
Tested-by: Kostya Porotchkin <Kostya.Porotchkin@cavium.com>
Reviewed-by: Kostya Porotchkin <Kostya.Porotchkin@cavium.com>
---
 arch/arm/dts/armada-8k.dtsi    |  2 +-
 arch/arm/dts/armada-cp110.dtsi | 14 +++++++-------
 arch/arm/dts/cn9130-db.dtsi    |  2 --
 arch/arm/dts/cn9130.dtsi       |  2 +-
 arch/arm/dts/cn9131-db.dtsi    |  2 ++
 arch/arm/dts/cn9132-db.dtsi    |  2 ++
 6 files changed, 13 insertions(+), 11 deletions(-)

diff --git a/arch/arm/dts/armada-8k.dtsi b/arch/arm/dts/armada-8k.dtsi
index a708ec83a2..d777537d5c 100644
--- a/arch/arm/dts/armada-8k.dtsi
+++ b/arch/arm/dts/armada-8k.dtsi
@@ -12,7 +12,7 @@
 #define CP110_BASE				(CP110_BASE_OFFSET + \
 	((CP110_NUM % 2) * CP110_SPACE_SIZE))
 
-#define CP110_PCIE_MEM_SIZE			(0xf00000)
+#define CP110_PCIE_MEM_SIZE(iface)	(0xf00000)
 #define CP110_PCIEx_CPU_IO_BASE(iface)		\
 	(0xf9000000 + (CP110_NUM % 2) * 0x4010000 + (iface) *  0x10000)
 #define CP110_PCIEx_CPU_MEM_BASE(iface)		\
diff --git a/arch/arm/dts/armada-cp110.dtsi b/arch/arm/dts/armada-cp110.dtsi
index 7668f06e8b..90820fad27 100644
--- a/arch/arm/dts/armada-cp110.dtsi
+++ b/arch/arm/dts/armada-cp110.dtsi
@@ -14,7 +14,7 @@
 #define CP110_PCIEx_REG0_BASE(iface)	\
 	(CP110_BASE + 0x600000 + (iface) * 0x20000)
 #define CP110_PCIEx_REG1_BASE(iface)	\
-	(CP110_PCIEx_CPU_MEM_BASE(iface) + CP110_PCIE_MEM_SIZE)
+	(CP110_PCIEx_CPU_MEM_BASE(iface) + CP110_PCIE_MEM_SIZE(iface))
 / {
 	CP110_NAME {
 		#address-cells = <2>;
@@ -278,8 +278,8 @@
 				/* non-prefetchable memory */
 				CP110_PCIE_BUS_MEM_CFG U64_TO_U32_H(CP110_PCIEx_BUS_MEM_BASE(0))
 				U64_TO_U32_L(CP110_PCIEx_BUS_MEM_BASE(0)) U64_TO_U32_H(CP110_PCIEx_CPU_MEM_BASE(0))
-				U64_TO_U32_L(CP110_PCIEx_CPU_MEM_BASE(0)) U64_TO_U32_H(CP110_PCIE_MEM_SIZE)
-				U64_TO_U32_L(CP110_PCIE_MEM_SIZE)>;
+				U64_TO_U32_L(CP110_PCIEx_CPU_MEM_BASE(0))
+				U64_TO_U32_H(CP110_PCIE_MEM_SIZE(0)) U64_TO_U32_L(CP110_PCIE_MEM_SIZE(0))>;
 			num-lanes = <1>;
 			status = "disabled";
 		};
@@ -305,8 +305,8 @@
 				/* non-prefetchable memory */
 				CP110_PCIE_BUS_MEM_CFG U64_TO_U32_H(CP110_PCIEx_BUS_MEM_BASE(1))
 				U64_TO_U32_L(CP110_PCIEx_BUS_MEM_BASE(1)) U64_TO_U32_H(CP110_PCIEx_CPU_MEM_BASE(1))
-				U64_TO_U32_L(CP110_PCIEx_CPU_MEM_BASE(1)) U64_TO_U32_H(CP110_PCIE_MEM_SIZE)
-				U64_TO_U32_L(CP110_PCIE_MEM_SIZE)>;
+				U64_TO_U32_L(CP110_PCIEx_CPU_MEM_BASE(1)) U64_TO_U32_H(CP110_PCIE_MEM_SIZE(1))
+				U64_TO_U32_L(CP110_PCIE_MEM_SIZE(1))>;
 			num-lanes = <1>;
 			status = "disabled";
 		};
@@ -332,8 +332,8 @@
 				/* non-prefetchable memory */
 				CP110_PCIE_BUS_MEM_CFG U64_TO_U32_H(CP110_PCIEx_BUS_MEM_BASE(2))
 				U64_TO_U32_L(CP110_PCIEx_BUS_MEM_BASE(2)) U64_TO_U32_H(CP110_PCIEx_CPU_MEM_BASE(2))
-				U64_TO_U32_L(CP110_PCIEx_CPU_MEM_BASE(2)) U64_TO_U32_H(CP110_PCIE_MEM_SIZE)
-				U64_TO_U32_L(CP110_PCIE_MEM_SIZE)>;
+				U64_TO_U32_L(CP110_PCIEx_CPU_MEM_BASE(2)) U64_TO_U32_H(CP110_PCIE_MEM_SIZE(2))
+				U64_TO_U32_L(CP110_PCIE_MEM_SIZE(2))>;
 			num-lanes = <1>;
 			status = "disabled";
 		};
diff --git a/arch/arm/dts/cn9130-db.dtsi b/arch/arm/dts/cn9130-db.dtsi
index a5a51ce898..f844e1705f 100644
--- a/arch/arm/dts/cn9130-db.dtsi
+++ b/arch/arm/dts/cn9130-db.dtsi
@@ -130,6 +130,4 @@
 
 &cp0_pcie0 {
 	status = "okay";
-		/* non-prefetchable memory */
-	ranges =<0x82000000 0 0xc0000000 0 0xc0000000 0 0x2000000>;
 };
diff --git a/arch/arm/dts/cn9130.dtsi b/arch/arm/dts/cn9130.dtsi
index 9c3a5891c9..efed08a65c 100644
--- a/arch/arm/dts/cn9130.dtsi
+++ b/arch/arm/dts/cn9130.dtsi
@@ -21,7 +21,7 @@
 #define CP110_BASE			(CP110_BASE_OFFSET + \
 						(CP110_NUM * CP110_SPACE_SIZE))
 
-#define CP110_PCIE_MEM_SIZE		(0xf00000)
+#define CP110_PCIE_MEM_SIZE(iface)	((iface == 0) ? 0x1ff00000 : 0xf00000)
 #define CP110_PCIE_BUS_MEM_CFG		(0x82000000)
 
 /* CP110-0 Settings */
diff --git a/arch/arm/dts/cn9131-db.dtsi b/arch/arm/dts/cn9131-db.dtsi
index f5c460f752..9a75abbe33 100644
--- a/arch/arm/dts/cn9131-db.dtsi
+++ b/arch/arm/dts/cn9131-db.dtsi
@@ -7,6 +7,7 @@
 
 #undef CP110_NAME
 #undef CP110_NUM
+#undef CP110_PCIE_MEM_SIZE
 #undef CP110_PCIEx_CPU_IO_BASE
 #undef CP110_PCIEx_CPU_MEM_BASE
 #undef CP110_PCIEx_BUS_IO_BASE
@@ -15,6 +16,7 @@
 /* CP110-1 Settings */
 #define CP110_NAME			cp1
 #define CP110_NUM			1
+#define CP110_PCIE_MEM_SIZE(iface)	(0xf00000)
 #define CP110_PCIEx_CPU_IO_BASE(iface)	(0xf8030000 + (iface) * 0x10000)
 #define CP110_PCIEx_CPU_MEM_BASE(iface)	(0xe2000000 + (iface) * 0x1000000)
 #define CP110_PCIEx_BUS_IO_BASE(iface)	(CP110_PCIEx_CPU_IO_BASE(iface))
diff --git a/arch/arm/dts/cn9132-db.dtsi b/arch/arm/dts/cn9132-db.dtsi
index b38c964c91..52c29e0816 100644
--- a/arch/arm/dts/cn9132-db.dtsi
+++ b/arch/arm/dts/cn9132-db.dtsi
@@ -7,6 +7,7 @@
 
 #undef CP110_NAME
 #undef CP110_NUM
+#undef CP110_PCIE_MEM_SIZE
 #undef CP110_PCIEx_CPU_IO_BASE
 #undef CP110_PCIEx_CPU_MEM_BASE
 #undef CP110_PCIEx_BUS_IO_BASE
@@ -15,6 +16,7 @@
 /* CP110-2 Settings */
 #define CP110_NAME			cp2
 #define CP110_NUM			2
+#define CP110_PCIE_MEM_SIZE(iface)	(0xf00000)
 #define CP110_PCIEx_CPU_IO_BASE(iface)	(0xf8060000 + (iface) *  0x10000)
 #define CP110_PCIEx_CPU_MEM_BASE(iface)	(0xe5000000 + (iface) *  0x1000000)
 #define CP110_PCIEx_BUS_IO_BASE(iface)	(CP110_PCIEx_CPU_IO_BASE(iface))
-- 
2.22.0

